homeassistant/components/sonos/number.py

```diff
@@ -92,9 +110,11 @@ def poll_state(self) -> None:
     @soco_error()
     def set_native_value(self, value: float) -> None:
         """Set a new value."""
-        setattr(self.soco, self.level_type, value)
+        from_number = LEVEL_FROM_NUMBER.get(self.level_type, int)
```

Comment:
Converting all the number fields to `int` before re-casting back to `float` (and vice versa) _seems_ okay. Did you notice any cases where this could be risky?

Reply:
All currently existing entities have step 1 as far as I can tell.